osf/models/analytics.py

```diff
@@ -1,25 +1,151 @@
-    total = models.PositiveIntegerField()
-    unique = models.PositiveIntegerField()
+    total = models.PositiveIntegerField(default=0)
+    unique = models.PositiveIntegerField(default=0)
+
+    @classmethod
+    def migrate_from_modm(cls, modm_obj):
+        # TODO do this
+        pass
+
+    @staticmethod
+    def clean_page(page):
+        return page.replace(
+            '.', '_'
+        ).replace(
+            '$', '_'
+        )
+
+    @classmethod
+    def update_counter(cls, page, node_info):
+        cleaned_page = cls.clean_page(page)
+        date = timezone.now()
+        date_string = date.strftime('%Y/%m/%d')
+        visited_by_date = session.data.get('visited_by_date', {'date': date_string, 'pages': []})
+
+        with transaction.atomic():
+            model_instance, created = cls.objects.select_for_update().get_or_create(_id=cleaned_page)
+
+            # if they visited something today
+            if date_string == visited_by_date['date']:
+                # if they haven't visited this page today
+                if cleaned_page not in visited_by_date['pages']:
+                    # if the model_instance has today in it
+                    if date_string in model_instance.date.keys():
+                        # increment the number of unique visitors for today
+                        model_instance.date[date_string]['unique'] += 1
+                    else:
+                        # set the number of unique visitors for today to 1
+                        model_instance.date[date_string] = dict(unique=1)
+            # if they haven't visited something today
+            else:
+                # set their visited by date to blank
+                visited_by_date['date'] = date_string
+                visited_by_date['pages'] = []
+                # if the model_instance has today in it
+                if date_string in model_instance.date.keys():
+                    # increment the number of unique visitors for today
+                    model_instance.date[date_string]['unique'] += 1
+                else:
+                    # set the number of unique visitors to 1
+                    model_instance.date[date_string] = dict(unique=1)
+
+            # update their sessions
+            visited_by_date['pages'].append(cleaned_page)
+            session.data['visited_by_date'] = visited_by_date
+
+            if date_string in model_instance.date.keys():
+                if 'total' not in model_instance.date[date_string].keys():
+                    model_instance.date[date_string].update(total=0)
+                model_instance.date[date_string]['total'] += 1
+            else:
+                model_instance.date[date_string] = dict(total=1)
+
+            # if a download counter is being updated, only perform the update
+            # if the user who is downloading isn't a contributor to the project
+            page_type = cleaned_page.split(':')[0]
+            if page_type == 'download' and node_info:
```

Comment:
It appears that we only call this for file downloads so I'm not sure why this is here but I left it because I wasn't 100% sure.